app/Http/Auth/Controllers/LoginController.php

```diff
@@ -22,15 +22,21 @@ public function __construct()
         $this->middleware('guest')->except('logout');
     }
 
-    public function showLoginForm()
+    public function showLoginForm(Request $request)
     {
         $previous = url()->previous();
 
         if (Str::finish($previous, '/') === Str::finish(url('/'), '/')) {
             $previous = route('products.index');
         }
 
-        session()->flash('next', request('next') ?? $previous);
+        session()->flash('next', $previous);
+
+        if ($request->get('next') !== null &&
```

Comment:
It isn't immediately clear what this does. Let's improve readability by refactoring this to a dedicated function with a good name, perhaps `onlyAllowSpatieUrls()`. In that function avoid using `&&`, but use early returns 👍